### ISSUE TITLE
SQL Server Compact 4.0 BulkCopy throws System.StackOverflowException for 900 items (inserting one by one is working correctly) 

### ISSUE THREAD

**Issue by alexey-leonovich** [linq2db/linq2db#4574]
### Describe your issue

I have an entity with two Guid fields. If I insert 900 sample entites one by one it is working without any problem.
If I insert same entites with BulkCopy - System.StackOverflowException is thrown.
Less items (850 for example) are inserting with no problem even by BulkCopy.

### Steps to reproduce

1. Create .Net Framework 4.7.2 console application
2. Reference linq2db (I used latest version 5.4.1), Microsoft.SqlServer.Compact and Newtonsoft.Json nugets
3. Copy sample code below to Program.cs
4. Copy test data to application output directory - [test.txt](https://github.com/user-attachments/files/16248957/test.txt)
5. Run application - get an exception

```c#
using LinqToDB.Data;
using LinqToDB.Mapping;
using LinqToDB;
using System;
using System.IO;
using System.Linq;
using LinqToDB.DataProvider.SqlCe;
using Newtonsoft.Json;

namespace SqlCeStackOverflowBugTest
{
    [Table(Name = "ReservesTables")]
    internal sealed class ReservesTablesItem
    {
        [Column(CanBeNull = false)]
        public Guid ReserveId { get; set; }

        [Column(CanBeNull = false)]
        public Guid TableId { get; set; }
    }

    internal class Program
    {
        static void Main(string[] args)
        {
            using (var db = new DataConnection(SqlCeTools.GetDataProvider(), "Data Source=test.sdf"))
            {
                SqlCeTools.CreateDatabase("test.sdf");
                db.CreateTable<ReservesTablesItem>();
                var items = File.ReadAllText("test.txt")
                        .Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries)
                        .Select(JsonConvert.DeserializeObject<ReservesTablesItem>).ToList();
                //single insert is working correctly
                //foreach (var item in items)
                //{
                //    db.Insert(item);
                //}
                //bulk insert throws System.StackOverflowException
                db.BulkCopy(items);
            }
        }
    }
}
```

### Environment details

Linq To DB version: *5.4.1*

Database (with version): *SQL Server Compact 4.0*

ADO.NET Provider (with version): *Microsoft.SqlServer.Compact 4.0.8876.1*

Operating system: *Windows 10 22H2*

.NET Version: *.Net Framework 4.7.2*

**Comment by alexey-leonovich** [comment]
@MaceWindu Were you able to reproduce this bug? Do you need any additional information from me?

**Comment by MaceWindu** [comment]
Yes, it comes from SQLCE itself so probably nothing we can do here. You should limit batch size:

```cs
db.BulkCopy(new BulkCopyOptions()
{ 
	MaxBatchSize = 500
}, items);
```

**Comment by MaceWindu** [comment]
I think we have SQL limits in bulk API implementation, so we could try to enforce them on SQL CE implementation. Just need to find when it breaks. 

Same issue : https://stackoverflow.com/questions/37472026/system-stackoverflowexception-occurred-in-system-data-sqlserverce-dll

**Comment by alexey-leonovich** [comment]
BTW. Breaking code sample was ported some time ago from BLToolkit to linq2db. And BLToolkit.Data.DbManager.InsertBatch() was working for many years without any issues. Looks like BLToolkit had such SQL limit check but linq2db doesn't. Maybe that can help you.

**Comment by MaceWindu** [comment]
Well, it was doing it in per-record mode without batching multiple records in single query https://github.com/igor-tkachev/bltoolkit/blob/089fbf3298ac57e46ae919f8ed17a1091db2412e/Source/Data/DataProvider/DataProviderBase.cs#L521-L522

**Comment by MaceWindu** [comment]
What I find interesting is that it doesn't crash in .net core

**Comment by alexey-leonovich** [comment]
That is really interesting, I've tested 500_000 records on .NET 6.0 and they were inserted correctly. Same ADO .NET Provider (System.Data.SqlServerCe.dll compiled for .NET Framework 4.0), same ORM (linq2db.dll compiled for .Net Standard 2.0), same code - but it crashes on one runtime and is working on another one..
